mypy/myunit/__init__.py

```diff
@@ -112,7 +112,7 @@ def __init__(self, name: str, suite: 'Optional[Suite]' = None,
         self.name = name
         self.suite = suite
         self.old_cwd = None  # type: Optional[str]
-        self.tmpdir = None  # type: Optional[tempfile.TemporaryDirectory]
+        self.tmpdir = None  # type: Optional[tempfile.TemporaryDirectory[str]]
```

Comment:
Totally off-topic: this would be a case where if you converted it to PEP 526 style (`tmpdir: Optional[tempfile.TemporaryDirectory[str]]` at the class level) it would fail at runtime, because the real tempfile module doesn't declare TemporaryDirectory as generic -- but Łukasz's PEP 563 would save the day...

Reply:
Yes, this is a good case for PEP 563.

Reply:
This has surfaced in #5087 during self-check:
> mypy/test/data.py:232: error: "TemporaryDirectory" expects no type arguments, but 1 given

How did that pass before?